stixcore/processing/publish.py

```diff
@@ -349,6 +352,116 @@ def send_mail_report(files):
             logger.error(f"Error: unable to send report email: {e}")
 
 
+def update_ephemeris_headers(fits_file, spice):
+    product = Product(fits_file)
+    if product.level in ['L1', 'L2']:
+        ephemeris_headers = \
+            spice.get_fits_headers(start_time=product.utc_timerange.start,
+                                   average_time=product.utc_timerange.center)
+        with fits.open(fits_file, 'update') as f:
+            for hdu in f:
+                now = datetime.now().isoformat(timespec='milliseconds')
+                hdu.header['HISTORY'] = f'updated ephemeris header with latest kernel at {now}'
+                hdu.header.update(ephemeris_headers)
+                # just update the first prima HDU
+                break
+        logger.info(f"updated ephemeris headers of {fits_file}")
+
+
+def addBSDComment(p, rid_lut):
+    try:
+        parts = p.name.split('_')
+        if len(parts) > 5:
+            rid = parts[5].replace(".fits", "")
+            rid = int(rid.split('-')[0])
+            reqeust = rid_lut.loc[rid]
+            reason = " ".join(np.atleast_1d(reqeust['description']))
+            c_entry = f"BSD request id: '{rid}' reason: '{reason}'"
+            fits.setval(p, "COMMENT", value=c_entry)
+            return c_entry
+    except Exception as e:
+        logger.debug(e, exc_info=True)
+        logger.info(f"no BSD request comment added nothing found in LUT for rid: {rid}")
+    return ''
+
+
+def addHistory(p, name):
+    time_formated = datetime.now().isoformat(timespec='milliseconds')
+    h_entry = f"published to ESA SOAR as '{name}' on {time_formated}"
+    fits.setval(p, "HISTORY", value=h_entry)
+
+
+def copyFile(scp, p, target_dir, add_history=True):
```

Comment:
```suggestion
def copy_file(scp, p, target_dir, add_history=True):
```

Also add short docstring